### ISSUE TITLE
pipx fails to build osxphotos on Python 3.12

### ISSUE THREAD

**Issue by lxgr** [RhetTbull/osxphotos#1254]
**Describe the bug**
`pipx install osxphotos` consistently fails on python 3.12, but succeeds on 3.11, on macOS Sonoma.

**To Reproduce**
Command: `pipx install --python python3.12 osxphotos`
```
Collecting pyobjc-framework-Cocoa<9.0,>=7.3 (from osxphotos)
  Using cached pyobjc-framework-Cocoa-7.3.tar.gz (3.5 MB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'

PIP STDERR
----------
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [9 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/0f/_60rf_ys4z7d7s0nw3_b56bh0000gn/T/pip-install-zl_dr3um/pyobjc-framework-cocoa_06a4b62f59f94ef7b02153aa87afd3de/setup.py", line 23, in <module>
          Extension(
        File "/private/var/folders/0f/_60rf_ys4z7d7s0nw3_b56bh0000gn/T/pip-install-zl_dr3um/pyobjc-framework-cocoa_06a4b62f59f94ef7b02153aa87afd3de/pyobjc_setup.py", line 459, in Extension
          "-DPyObjC_BUILD_RELEASE=%02d%02d" % (tuple(map(int, os_level.split("."))))
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      TypeError: not enough arguments for format string
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.
```

**Desktop (please complete the following information):**
 - OS: macOS Sonoma 14.0
 - osxphotos version: n/a (latest)

**Comment by RhetTbull** [comment]
osxphotos does not run on 3.12. The documentation states is supports 3.9 - 3.11.  See #1245

**Comment by lxgr** [comment]
Oh, sorry, I missed that!

Is there a way to let pip or pipx know about the supported Python versions?

**Comment by RhetTbull** [comment]
pip uses whichever python version is installed. It is actually a python package (you can also run it via `python3 -m pip install ...`).  I think with pipx, you can use `pipx install osxphotos --python /path/to/python`  

How did you install python? You can have multiple python versions if needed. 

I will get osxphotos updated for 3.12 (I first need to update some of its dependencies which I'm also the author of) but am in the middle of another project at the moment.

**Comment by lxgr** [comment]
Ah, sorry if that was not clear – I was able to already resolve this by using `pipx install --python python3.11 osxphotos`.

I was just wondering if it was possible to include this metadata in PyPI somehow, but this seems to be an unresolved problem for pipx as well: https://github.com/pypa/pipx/issues/387

Seems like the only possible fix would be to possibly explicitly specify the version in the [installation instructions](https://github.com/RhetTbull/osxphotos#installation), but that would needlessly force people to install a Python despite their current one being compatible (e.g. in case they have 3.10 locally).

But maybe a hint to use `--python <version>` could help?

**Comment by RhetTbull** [comment]
I think this is fixed in v0.64.2
